identity/backend/src/main/java/io/camunda/identity/usermanagement/service/UserService.java

```diff
@@ -40,64 +45,66 @@ public UserService(
     this.passwordEncoder = passwordEncoder;
   }
 
-  public CamundaUser createUser(final CamundaUserWithPassword userWithCredential) {
+  public CamundaUser createUserFailIfExists(
```

Comment:
❓ This feels like bike shedding again! however, as a caller of this method what should I do with the `FailIfExists` part? Is there an action I can take? Could we instead use the old name but just declare that it throws an exception on failure?

Reply:
I introduced this to cover both cases as we have the situation with the configured demo user(s) trying to be created but won't succeed if they already exist ... given our setup will likely run with a more steady db than H2 in some cases, I believe this should be covered.

Then on the other hand, with some distance, I see how this could be solved differently. I wonder if we should instead have logic in the user initializer that first checks if the users exist and acts accordingly. Would you agree to go this way instead? I'd prefer it on a second thought now :)

Reply:
as discussed during our call, I've switched to:
- `createUser` method in the service which throws an exception "User already exists" 
- DefaultUserInitializer checks if users exist and only tries to create them if not